redis/src/sentinel.rs

```diff
@@ -742,6 +742,26 @@ impl SentinelClient {
     }
 }
 
+/// All options for async connections to redis
+#[cfg(feature = "aio")]
+#[cfg_attr(docsrs, doc(cfg(feature = "aio")))]
+#[cfg(any(feature = "tokio-comp", feature = "async-std-comp"))]
+pub struct AsyncConnectionOptions {
+    /// Timeouts for async connections to redis
+    pub timeouts: Option<AsyncConnectionOptionsForTimeout>,
+}
+
+/// Timeouts for async connections to redis
+#[cfg(feature = "aio")]
+#[cfg_attr(docsrs, doc(cfg(feature = "aio")))]
+#[cfg(any(feature = "tokio-comp", feature = "async-std-comp"))]
+pub struct AsyncConnectionOptionsForTimeout {
+    /// Maximum time to wait for a response from the server
+    pub response_timeout: std::time::Duration,
```

Comment:
please move these fields into `AsyncConnectionOptions` - there's no need for a nested structure.
also, instead of making these fields public, please use the builder pattern, so that if we change the options' internal structure it won't be a breaking change.

Reply:
Will do

Reply:
Done

Reply:
Thanks, but sorry for not being explicit - can you split the with_timeouts function, in case a user is only interested in one timeout type?